### DISCUSSION TITLE
Service is running fine but it doesn't logging to specified log file!

### DISCUSSION THREAD

**Discussion by persian85** [DNSCrypt/dnscrypt-proxy#1949] [Q&A]
I just installed it on my fresh arch-linux exactly like arch wiki. It is working fine but it doesn't logging to the specified log file. I also changed the log file path to current user home folder but still log file is empty!  
log level is 2
log file path is correct
use syslog is false (even with true nothing changes)

[dnscrypt-proxy.txt](https://github.com/DNSCrypt/dnscrypt-proxy/files/7593775/dnscrypt-proxy.txt)

**Comment by jedisct1** [comment]
What happens when you start it from the command line? (with `sudo dnscrypt-proxy -config /etc/dnscrypt-proxy/dnscrypt-proxy.toml`)?

Do you see logs printed to the console?

**Reply by persian85** [reply]
Thank you for your reply.
So I stopped dnscrypt-proxy service with systemctl and ran the service like you mentioned. after that nothing printed on the console but now I can see that it's logging on log file!
So what do you suggest? 
I enabled the service via systemctl. Should I use another solution?

**Comment by ianbashford** [comment]
I can see in the default `dnscrypt-proxy.toml` on arch linux they have 
```use_syslog = true```

You'll see the entries by using 
```journalctl -u dnscrypt-proxy```

If you want it in the file rather than the journal, then set
```use_syslog = false```

And it'll start using the file as you expect

**Reply by persian85** [reply]
As I mentioned in my comment I already changed that line to false but nothing changed.

**Reply by jedisct1** [reply]
Try putting random garbage in your configuration file. If the server still starts, the configuration file you are looking at is not the one actually loaded.

**Reply by persian85** [reply]
Yes this is the config file that dnscrypt-proxy uses. I did some changes and it happens on service.

**Comment by ianbashford** [comment]
My arch linux is working as expected, so it's likely config or permissions, so I'd start by going for something minimal.
Below is a stripped down config file  -- my copy was in /tmp so I tested it by running

```  dnscrypt-proxy -config /tmp/dnscrypt-proxy-minimal.toml ```

This should just print up the screen as expected 
You can then introduce logging in /tmp , and then to your target folder, and use this to deduce where the issue is

```
listen_addresses = [':9954']

# log_file = '/var/log/dnscrypt-proxy/dnscrypt-proxy.log'
# log_file = '/tmp/minimal-dnscrypt-proxy.log'

use_syslog = false

ipv4_servers = true
dnscrypt_servers = true

[sources]

  ## An example of a remote source from https://github.com/DNSCrypt/dnscrypt-resolvers

  [sources.'public-resolvers']
    urls = ['https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v3/public-resolvers.md', 'https://download.dnscrypt.info/resolvers-list/v3/public-resolvers.md', 'https://ipv6.download.dnscrypt.info/resolvers-list/v3/public-resolvers.md', 'https://download.dnscrypt.net/resolvers-list/v3/public-resolvers.md']
    cache_file = '/var/cache/dnscrypt-proxy/public-resolvers.md'
    minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
    refresh_delay = 12
    prefix = ''

```

**Reply by persian85** [reply]
I found something that may help to resolve my issue. When I run dnscrypt-proxy manually I get error that couldn't find config file!
[FATAL] Unable to load the configuration file [dnscrypt-proxy.toml] -- Maybe use the -config command-line switch?

But i'm sure that config file is present in /etc/dnscrypt-proxy/dnscrypt-proxy.toml

Any idea?

**Reply by ianbashford** [reply]
Did you try the minimal config I laid out  in https://github.com/DNSCrypt/dnscrypt-proxy/discussions/1949#discussioncomment-1708824 ?

I'd use the ```-config ...``` switch if you're running from the command line as it's best to be explicit while you're debugging.  I think by default it's looking in the current directory, so it' can't find ```./dnscrypt-proxy.toml```

I notice on arch in ```/etc/systemd/system/multi-user.target.wants/dnscrypt-proxy.service```   that the ExecStart directive uses ```--config``` instead of ```-config```  
That seems to work correctly under systemctl but don't use that from a terminal -- just one ```-```

**Reply by persian85** [reply]
Yes I tried your minimal config with -config and it worked. As you said when service start with systemctl it shows that it uses --config to load the config file. (/usr/bin/dnscrypt-proxy --config /etc/dnscrypt-proxy/dnscrypt-proxy.toml)
What should I do to make it work when it starts with systemctl?

**Reply by ianbashford** [reply]
Well I'd start by moving ```/etc/dnscrypt-proxy/dnscrypt-proxy.toml```  to a safe spot, and put the minimal config in place -- then use systemctl to start the service, and check the file is still written - if you did use ```/tmp```, then remove the file (which will be owned by your user)  so that the system user (```dnscrypt-proxy```)  can write the file properly.
If it is, then you've identified an issue with your original ```dnscrypt-proxy.toml``` file and can debug the differences by adding them back.
If it isn't written, and you've not changed the file, then there's something else afoot... 

Take note of the permissions on /var/log/private/dnscrypt-proxy  which look like this on mine

```
ls -l /var/log/private/
drwxr-xr-x 2 dnscrypt-proxy dnscrypt-proxy 4096 Sep 14 19:14 dnscrypt-proxy
```
